xbmc/rendering/gl/RenderSystemGL.cpp

```diff
@@ -680,38 +672,24 @@ void CRenderSystemGL::ReleaseShaders()
 {
   if (m_pShader[SM_DEFAULT])
     m_pShader[SM_DEFAULT]->Free();
-  delete m_pShader[SM_DEFAULT];
```

Comment:
any reason why you don't release the smart pointer here? you can't call m_pShader[SM_DEFAULT]->Free() two times in a row but you allow this with this code.

Reply:
Yes that's a good point, let me fix that.